cmd/controller/app/options/options.go

```diff
@@ -285,6 +289,8 @@ func (s *ControllerOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.StringVar(&s.MetricsListenAddress, "metrics-listen-address", defaultPrometheusMetricsServerAddress, ""+
 		"The host and port that the metrics endpoint should listen on.")
+	fs.BoolVar(&s.EnablePprof, "enable-profiling", false, ""+
+		"Enable profiling for controller.")
```

Comment:
I notice that the webhook profiling is always-on: https://github.com/jetstack/cert-manager/pull/2450

I like your flag approach but we should probably add the same flag to the webhook too.
And we could consider making the profiling options available via a helm chart values.
Not necessarily here, but perhaps in some followup PRs.